bson/value_reader_test.go

```diff
@@ -26,1868 +26,2488 @@ var lorem []byte
 
 var testcstring = append(lorem, []byte{0x00}...)
 
-func TestValueReader(t *testing.T) {
-	type subtest struct {
-		name string
-		src  valueReaderByteSrc
+func TestValueReader_ReadBinary(t *testing.T) {
```

Comment:
@qingyang-hu , @matthewdale There were no logical changes to the tests. This is the update requested on the PR to avoid nesting subtests and keying test case parameters.